### PR TITLE
fix(ui-components): styling issues in CloseButton

### PR DESCRIPTION
**Pull request by huyenltnguyen**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

This PR:
- Fixes styling issues in the CloseButton component
- Improves the implementation of the component to ensure screen readers won't announce the `x` symbol to the users (screen readers are currently read the button as "times button", tested with VoiceOver)

## Screenshots

| | Before | After |
| --- | --- | --- |
| Light theme - at rest | Screenshot 2023-10-09 at 23 05 11 | Screenshot 2023-10-09 at 23 01 48 |
| Light theme - focused | Screenshot 2023-10-09 at 23 05 57 | Screenshot 2023-10-09 at 23 02 12 |
| Dark theme - at rest | Screenshot 2023-10-09 at 23 05 19 | Screenshot 2023-10-09 at 23 01 56 |
| Dark theme - focused | Screenshot 2023-10-09 at 23 05 34 | Screenshot 2023-10-09 at 23 02 04 |

### CODE REVIEW

File: tools/ui-components/src/close-button/close-button.tsx
Comment (Sboonny) [inline tools/ui-components/src/close-button/close-button.tsx:37]: We are using `:focus-visible` in `/learn` for these buttons, I will update the button component in https://github.com/freeCodeCamp/freeCodeCamp/pull/51443
```suggestion
    'focus-visible:ring',
    'focus-visible:ring-focus-outline-color',
```

File: tools/ui-components/src/close-button/close-button.tsx
Comment (Sboonny) [reply tools/ui-components/src/close-button/close-button.tsx:33]: True, it's controlled by the consumers, but I think it's safe to make that assumption, there is a general button component for when they need a button that doesn't close something, and we allow className, so they can add these classes for their unique cases. 

I am not experienced, so this may be common more than I think, so I am leaving this to your judgment. if you think this worth keeping, add the pseudo-element, because the styles aren't affecting it, because it doesn't have content, I think

File: tools/ui-components/src/close-button/close-button.tsx
Comment (Sboonny) [reply tools/ui-components/src/close-button/close-button.tsx:22]: the 24px were too big for the setting buttons, and moving it to 23px was perfect to mimic the old styles.

File: tools/ui-components/src/close-button/close-button.tsx
Comment (Sboonny) [reply tools/ui-components/src/close-button/close-button.tsx:33]: > we asked them to build a separate component themselves if we don't think the feature should be part of the library 

I do like that so much, there is a `CloseButton` in the `/learn`, and I won't be using this component in the `Modal`, because I don't need it. we build the component in `/learn`.

This way, we won't have to worry about maintaining the library with `react-i18next` in it, and if freeCodeCamp have to move to a different solution for i18n, we won't have to worry about it being here in the library.

We can merge this as is, and when the `Button` PR is merged, and the decoupling of `Link` and `Button` happen, we can create a PR that delete this component and move it to learn, this way the `active` that are used for links aren't used in the buttons, and we won't have to worry about maintaining translation library feature here.

---

<details>
<summary>Context of what I know about the future of the library</summary>

This is all the context that I have, I will focus the current implementation of the UI components library only on /learn, the future implementation will start to account for how to handle unique cases. 

Why is that? because I don't know which next milestone will be after we move away from bootstrap.

if we focus on `/news`, then we have to decide if we are moving `/news` to react or making web-components, aka `<fcc-button></fcc-button>`, and this library will act as wrapper for `/learn` usage.

if we focus on `/publish` and `/classroom`, we will have to account for these situations and if they do happen in there, but they would already have made their own features, and it will be the other way around, do we need the feature in `/classroom` and `/publish` in `/learn`? I personally think we won't.
</details>

File: tools/ui-components/src/close-button/close-button.tsx
Comment (Sboonny) [reply tools/ui-components/src/close-button/close-button.tsx:22]: I will re-investigate it in the `Button` PR 🤔, I remember an issue existing. 

And you are correct, if there are a set of buttons that don't have similar styles in the `Button`, it's better to add new styles and new value there, instead of creating one for this specific niche case, should have thought about it more thoroughly.

I will revert it back, and see which button it was and add something like `text-name` there.

**Review by Sboonny — APPROVED** [review]
LGTM

image

File: tools/ui-components/src/close-button/close-button.tsx
Comment (Sboonny) [reply tools/ui-components/src/close-button/close-button.tsx:33]: I have linked your point in ui-component issue, so it doesn't get lost easily. I am resolving this, because further discussion on how to apply this, may block this PR.

**Review by naomi-lgbt — APPROVED** [review]

**Pull request merged**
